api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1886,6 +1886,12 @@ func GetProcessPort(processNumber int, tls bool) int {
 // the primary address.
 func (cluster *FoundationDBCluster) GetFullAddressList(address string, primaryOnly bool, processNumber int) []ProcessAddress {
 	addrs := make([]ProcessAddress, 0, 2)
+
+	// If the address is already enclosed in brackets, remove them since they
+	// will be re-added automatically in the ProcessAddress logic.
+	if strings.HasPrefix(address, "[") && strings.HasSuffix(address, "]") {
```

Comment:
We could directly use `TrimPrefix` and `TrimSuffix` the complexity will be the same but we touch the string only once.